internal/translator/translate_test.go

```diff
@@ -31,7 +31,17 @@ func TestTranslateIngressRoute(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
-			hp, warnings := IngressRouteToHTTPProxy(ir)
+			hp, warnings, err := IngressRouteToHTTPProxy(ir)
+			if err != nil {
+				// Can't translate the IngressRoute at all
+				// errors.txt should have the error message.
+				errorDiff := cmp.Diff([]string{err.Error()}, tc.warnings)
```

Comment:
I'm not sure what this code is doing. I suggest doing this

1. isolated all the failure cases that can return an err value and handle them in a separate test.
2. With those isolated, if err is returned then this is a bug in your test; so you only need to compare warnings to the expected fixture, and hp to the expected fixture.

Reply:
Okay, thanks.

Reply:
Maybe I'm reviewing the wrong revision, but this is still todo.